src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,42 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String brand;
+    private String color;
+    private int speed;
+    private int doorsNum;
+    private Engine carEngine;
 
+    public Car(String brand, String color, int speed, int doorsNum, Engine carEngine) {
+        this.brand = brand;
+        this.color = color;
+        this.speed = speed;
+        this.doorsNum = doorsNum;
+        this.carEngine = carEngine;
+    }
+
+    public String getBrand() {
+        return brand;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public int getSpeed() {
+        return speed;
+    }
+
+    public int getDoorsNum() {
+        return doorsNum;
+    }
+
+    public Engine getCarEngine() {
+        return carEngine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        return new Car(brand, color, speed, doorsNum, carEngine);
```

Comment:
you will return new Car but with the same engine. Guess the original Car wouldn't like it